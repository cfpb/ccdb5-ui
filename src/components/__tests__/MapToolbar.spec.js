import configureMockStore from 'redux-mock-store';
import ReduxMapToolbar, {
  mapDispatchToProps,
  mapStateToProps,
  MapToolbar,
} from '../Map/MapToolbar';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';

/**
 *
 */
function setupEnzyme() {
  const props = {
    clearStates: jest.fn(),
    filteredStates: 'Texas',
    showComplaints: jest.fn(),
  };

  const target = shallow(<MapToolbar {...props} />);

  return {
    props,
    target,
  };
}

/**
 *
 * @param filteredStates
 */
function setupSnapshot(filteredStates) {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({
    query: {
      state: filteredStates,
    },
  });

  return renderer.create(
    <Provider store={store}>
      <ReduxMapToolbar />
    </Provider>
  );
}

describe('component: MapToolbar', () => {
  describe('initial state', () => {
    it('renders without crashing', () => {
      const target = setupSnapshot([]);
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders filtered states without crashing', () => {
      const target = setupSnapshot(['FL', 'TX']);
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });
    it('provides a way to call clearStates', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).clearStates();
      expect(dispatch.mock.calls.length).toEqual(1);
    });
    it('provides a way to call showComplaints', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).showComplaints();
      expect(dispatch.mock.calls.length).toEqual(1);
    });
  });

  describe('mapStateToProps', () => {
    it('maps state and props', () => {
      const state = {
        query: {
          state: ['LA', 'MS', 'ZZ'],
        },
      };
      const actual = mapStateToProps(state);
      expect(actual).toEqual({
        filteredStates: 'Louisiana, Mississippi',
      });
    });
  });

  describe('click actions', () => {
    it('allows the user to remove state filter', () => {
      const { target, props } = setupEnzyme();
      const button = target.find('a.clear');

      button.simulate('click');
      expect(props.clearStates).toHaveBeenCalled();
    });
    it('allows the user to view complaints by state', () => {
      const { target, props } = setupEnzyme();
      const button = target.find('a.list');

      button.simulate('click');
      expect(props.showComplaints).toHaveBeenCalled();
    });
  });
});
