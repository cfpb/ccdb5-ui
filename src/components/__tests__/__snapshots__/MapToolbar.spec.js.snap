// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: MapToolbar initial state renders filtered states without crashing 1`] = `
<div
  className="map-toolbar"
>
  <section
    className="state-heading"
  >
    <span>
      Florida, Texas
    </span>
    <a
      className="clear"
      onClick={[Function]}
    >
      <svg
        className="cf-icon-svg"
        viewBox="0 0 1000 1200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm261.8 692.2c19.4 19.6 19.3 51.3-.3 70.7-19.5 19.3-50.9 19.3-70.4 0L499.6 676.6 308 868.1c-19.6 19.4-51.3 19.3-70.7-.3-19.3-19.5-19.3-50.9 0-70.4l191.6-191.5-191.6-191.6c-19.3-19.8-18.9-51.4.9-70.7 19.4-18.9 50.4-18.9 69.8 0l191.6 191.5 191.5-191.5c19.6-19.4 51.3-19.3 70.7.3 19.3 19.5 19.3 50.9 0 70.4L570.3 605.9l191.5 191.5z"
        />
      </svg>
      Clear
    </a>
  </section>
  <section
    className="state-navigation"
  >
    <a
      className="list"
      href="#"
      onClick={[Function]}
    >
      View complaints for filtered states
    </a>
  </section>
</div>
`;

exports[`component: MapToolbar initial state renders without crashing 1`] = `
<div
  className="map-toolbar"
>
  <section
    className="state-heading"
  >
    <span>
      United States of America
    </span>
    <span>
      
    </span>
    
  </section>
  
</div>
`;
